BAC0/core/io/Read.py

```diff
@@ -414,32 +421,36 @@ def readMultiple(
                 raise UnknownObjectError("Unknown object {}".format(args))
             elif reason == "unknownProperty":
                 self._log.warning("Unknown property {}".format(args))
-
-                values.append("")
+                values.append("")  # type: ignore[arg-type]
                 return values
             else:
                 self._log.warning("No response from controller {}".format(reason))
-                values.append("")
+                values.append("")  # type: ignore[arg-type]
                 return values
 
-    def build_rp_request(self, args, arr_index=None, vendor_id=0, bacoid=None):
-        addr, obj_type, obj_inst, prop_id = args[:4]
-        vendor_id = vendor_id
-        bacoid = bacoid
+        return values
 
+    def __get_obj_type(self, obj_type: str, vendor_id) -> int:
         if obj_type.isdigit():
-            obj_type = int(obj_type)
+            return int(obj_type)
         elif "@obj_" in obj_type:
-            obj_type = int(obj_type.split("_")[1])
+            return int(obj_type.split("_")[1])
         elif not get_object_class(obj_type, vendor_id=vendor_id):
             raise ValueError("Unknown object type : {}".format(obj_type))
+        return obj_type  # type: ignore
 
-        obj_inst = int(obj_inst)
+    def build_rp_request(self, args: t.List[str], arr_index=None, vendor_id: int = 0, bacoid=None) -> ReadPropertyRequest:
+        addr, obj_type_str, obj_inst_str, prop_id_str = args[:4]
 
-        if prop_id.isdigit():
-            prop_id = int(prop_id)
-        elif "@prop_" in prop_id:
-            prop_id = int(prop_id.split("_")[1])
+        obj_type = self.__get_obj_type(obj_type_str, vendor_id)
+        obj_inst = int(obj_inst_str)
+
+        if prop_id_str.isdigit():
+            prop_id = int(prop_id_str)
```

Comment:
renamed variables because `mypy` doesn't like it when you re-assign a variable with a new type

Reply:
well... ok I guess.

Reply:
yeah not my favorite either